dotnet/src/SemanticKernel/KernelBuilder.cs

```diff
@@ -25,6 +31,9 @@ public sealed class KernelBuilder
     private IPromptTemplateEngine? _promptTemplateEngine;
     private readonly AIServiceCollection _aiServices = new();
 
+    private static bool _promptTemplateEngineInit = false;
```

Comment:
I suggest using more detailed naming for better understanding, for example `_templateEngineInitialized`.